File: CSharpFundamentals/CSharpOOPbasics/DefiningClassesEx/CarSalesman/Car.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CarSalesman
{
    public class Car
    {
        private string model;
        private Engine engine;
        private string weight;
        private string color;

        public Car(string model, Engine engine)
        {
            this.Model = model;
            this.Engine = engine;
            this.Weight = "n/a";
            this.Color = "n/a";
        }

        public string Color
        {
            get { return color; }
            set { color = value; }
        }

        public string Weight
        {
            get { return weight; }
            set { weight = value; }
        }

        public Engine Engine
        {
            get { return engine; }
            set { engine = value; }
        }

        public string Model
        {
            get { return model; }
            set { model = value; }
        }

        public override string ToString()
        {
            return $"{this.Model}:\n" +
                   $"  {this.Engine.Model}:\n" +
                   $"    Power: {this.Engine.Power}\n" +
                   $"    Displacement: {this.Engine.Displacement}\n" +
                   $"    Efficiency: {this.Engine.Efficiency}\n" +
                   $"  Weight: {this.Weight}\n" +
                   $"  Color: {this.Color}";
        }
    }
}

File: JavaFundamentals/JavaAdvanced/DefiningClasses_Ex/src/CarSalesman/Car.java
package CarSalesman;

public class Car {
    private String model;
    private Engine engine;
    private int weight = -1;
    private String color = "n/a";

    public Car(String model, Engine engine) {
        this.model = model;
        this.engine = engine;
    }

    public String getModel() {
        return model;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public int getWeight() {
        return weight;
    }

    public String getColor() {
        return color;
    }

    public Engine getEngine() {
        return engine;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(this.getModel())
                .append(":")
                .append(System.lineSeparator());
        sb.append(this.getEngine().getModel())
                .append(":")
                .append(System.lineSeparator());
        sb.append("Power: ")
                .append(this.getEngine().getPower())
                .append(System.lineSeparator());
        sb.append("Displacement: ")
                .append(this.getEngine().getDisplacement() == -1
                        ? "n/a"
                        : this.getEngine().getDisplacement())
                .append(System.lineSeparator());
        sb.append("Efficiency: ")
                .append(this.getEngine().getEfficiency())
                .append(System.lineSeparator());
        sb.append("Weight: ")
                .append(this.getWeight() == -1
                        ? "n/a"
                        : this.getWeight())
                .append(System.lineSeparator());
        sb.append("Color: ")
                .append(this.getColor());

        return sb.toString();
    }
}